backend/core/src/main/java/org/sonarsource/sonarlint/core/rules/RulesService.java

```diff
@@ -230,19 +230,17 @@ private ServerActiveRule tryConvertDeprecatedKeys(ServerActiveRule possiblyDepre
       var ruleKeyPossiblyWithDeprecatedRepo = RuleKey.parse(possiblyDeprecatedActiveRuleFromStorage.getRuleKey());
       var templateRuleKeyWithCorrectRepo = RuleKey.parse(ruleOrTemplateDefinition.get().getKey());
       var ruleKey = new RuleKey(templateRuleKeyWithCorrectRepo.repository(), ruleKeyPossiblyWithDeprecatedRepo.rule()).toString();
-      //TODO Replace empty list with proper one.
       return new ServerActiveRule(ruleKey, possiblyDeprecatedActiveRuleFromStorage.getSeverity(), possiblyDeprecatedActiveRuleFromStorage.getParams(),
-        ruleOrTemplateDefinition.get().getKey(), Collections.emptyList());
+        ruleOrTemplateDefinition.get().getKey(), possiblyDeprecatedActiveRuleFromStorage.getOverriddenImpacts());
     } else {
       ruleOrTemplateDefinition = rulesRepository.getRule(connectionId, possiblyDeprecatedActiveRuleFromStorage.getRuleKey());
       if (ruleOrTemplateDefinition.isEmpty()) {
         // The rule is not known among our loaded analyzers, so return it untouched, to let calling code take appropriate decision
         return possiblyDeprecatedActiveRuleFromStorage;
       }
-      //TODO Replace empty list with proper one.
       return new ServerActiveRule(ruleOrTemplateDefinition.get().getKey(), possiblyDeprecatedActiveRuleFromStorage.getSeverity(),
         possiblyDeprecatedActiveRuleFromStorage.getParams(),
-        null, Collections.emptyList());
+        null, possiblyDeprecatedActiveRuleFromStorage.getOverriddenImpacts());
     }
```

Comment:
This method is very similar to the one in AnalysisService, maybe there is a way to keep only one

Reply:
How about making it static on RuleService and using it on AnalysisService?